### ISSUE TITLE
Pasting settings to multiple images doesn't work reliably in the filmstrip

### ISSUE THREAD

**Issue by maboleth** [darktable-org/darktable#16116]
### Describe the bug

It doesn’t matter if you use ctrl-shift+V or just ctrl+V DT won’t paste anything on multiple selected images if your mouse is not positioned on the filmstrip in Darkroom - _and_ exactly on the selected ones.

1. If your mouse is away from the filmstrip it won’t paste anything, regardless of your selection.
2. If your mouse is on the selected image(s) it will paste the settings to all selected images
3. Sometimes you have to do this 2 or 3 times for DT to execute Paste to all selected.
4. It gets even weirder if you forget that hovered image gets priority over selected ones - you need to make sure your mouse pointer is positioned exactly on the selected ones.

This makes even weirder if you opt for CTRL+SHIFT+V because you need to position your mouse to agree and press "OK". The workaround is to have your mouse on the filmstrip prior executing ctrl+shift+v. Otherwise it won't work, even though DT will seem like it successfully pasted the settings - there's no comment that it failed.

### Steps to reproduce

1. Edit one image
2. Copy all desired settings
3. Select multiple images you want those settings to be applied on in the filmstrip of Darkroom
4. If your mouse is not on the filmstrip - ctrl+v/ctrl+shift+v will not do anything - no message about failed attempts either.

### Expected behavior

Paste always working, regardless of your mouse pointer.

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.6

### What OS are you using?

Linux

### What is the version of your OS?

Arch rolling

### Are you using OpenCL GPU in darktable?

None

**Comment by pehar1** [comment]
I think you are mainly complaining about the act-on feature. This has been discussed in depth several times in the past. Please search here and/or in pixls forum for the discussions. Some people hate this feature, others (like me) love it because it can be extremely powerful.

**Comment by maboleth** [comment]
> I think you are mainly complaining about the act-on feature. This has been discussed in depth several times in the past. Please search here and/or in pixls forum for the discussions. Some people hate this feature, others (like me) love it because it can be extremely powerful.

Just to be clear - I'm not talking about hovered thumbnail image having greater priority over your selected ones. I understand that.

This is about Paste not working entirely if your mouse cursor is not in the filmstrip region in Darkroom.
Not only it does not work, DT doesn't guide you, there's no warning or anything. It appears as if Paste worked, but it did not. 
It took me a bit to figure this out. We talked about this in Pixls forum yesterday.

**Comment by gi-man** [comment]
The copy/paste is part of the lighttable module. The darkroom is to process one image. I think your expectations are incorrect.

**Comment by TurboGit** [comment]
> This is about Paste not working entirely if your mouse cursor is not in the filmstrip region in Darkroom, even when you have selected all your images you want to have the settings Pasted.

And what do you propose? Pasting in main darkroom part? If you are not over the filmstrip in which image(s) do you expect the paste to be done? Please clarify.

**Comment by maboleth** [comment]
> If you are not over the filmstrip in which image(s) do you expect the paste to be done?

Well I've already selected the images, DT knows that I have a selection.

If I am able to select multiple images in Darkroom _and_ copy the settings _and_ paste them, I expect it to work regardless of where my mouse is positioned. Especially since DT doesn't give me any error or warning message that my pasting didn't succeed. 

From my perspective, I have selected the images, I've copied the settings, DT offered me to paste - why didn't that work?

At the very worst I think there should be some kind of a message that pasting failed due reason X or Y.
At the very best, DT should be able to "know" that I have selected the images in the filmstrip and want to paste the settings, so paste it on the selection, regardless of my mouse position.

As far as I know, there's no different copy or paste features, meaning there's no overlap of different variables in Darkroom that you can copy/paste. Whenever you use CTRL+SHIFT+C and CTRL+V the program knows exactly what are you trying to achieve, so should just apply these to a selection.

**Comment by TurboGit** [comment]
> From my perspective, I have selected the images, I've copied the settings, DT offered me to paste - why didn't that work?

You have selected the settings from the filmstrip while hovering an image. Right? So the pasting must be done also from filmstrip. I'm not sure to fully understand your issue. Are you proposing to be able to paste onto the selected images in the filmstrip even if your mouse is over the darkroom central area?

**Comment by pehar1** [comment]
> DT should be able to "know" that I have selected the images in the filmstrip and want to paste the settings, so paste it on the selection, **regardless of my mouse position**.

And that contradicts the act-on feature. For hovering the mouse over the central view I expect pasting acts on the image currently edited. And this is what happens (just checked again with 4.7.0+240~g930d9c0996, and it works).

**Comment by maboleth** [comment]
> You have selected the settings from the filmstrip while hovering an image. Right? So the pasting must be done also from filmstrip. I'm not sure to fully understand your issue. 

Alright, let's say I selected 3 images from the filmstrip that are from the same series. I edit one in Darkroom. I copy the settings and Paste it to other 2. I'm still editing in Darkroom and I'm using keyboard shortcuts.

Currently I need to move my mouse over the filmstrip every time I want to Paste the settings.

My proposal would be to keep editing and copy/pasting on-fly, seamlessly. I'm not touching the Filmstrip - **the images are already selected by me** and I don't want to change there anything. 

I'm focused on editing with my mouse, while copy/pasting by keyboard over my selection, wherever I need it.

> Are you proposing to be able to paste onto the selected images in the filmstrip even if your mouse is over the darkroom central area?

In a nutshell, yes. But please see my response above.

The hovering remains the same - if the user hovers any thumbnail in the Filmstrip, it also get Pasted. But when there's no hovering, just the selection, I think DT should Paste the settings there, no questions asked, regardless of user's mouse position.

At the very least, DT should inform the user that Paste failed due to mouse not being positioned over the selected thumbnails in the Filmstrip. But I hope that can be changed.

Thanks.

**Comment by maboleth** [comment]
> And that contradicts the act-on feature. For hovering the mouse over the central view I expect pasting acts on the image currently edited. And this is what happens (just checked again with 4.7.0+240~g930d9c0996, and it works).

That shouldn't be contradicting.

If you are hovering a thumbnail in the Filmstrip, that thumbnail still gets a priority - together with other selected ones.
That should remain the same.

I'm proposing that without hovering, with just the selection (that you already made in the Filmstrip before) - DT should apply Paste to all of selected images. That's all.

**Comment by wpferguson** [comment]
@maboleth If we change this to work the way you think it should, assuming it can even be done without a massive rewrite of the selection system, then it breaks the behavior for all of the users who use the act on workflow.  To make it "switchable" would still probably require lots of changes.

The act on images discussion has been raised numerous times with no consensus to a solution.

However, have you thought about using the act on behavior.  Using it you could hover over an image in the film strip, do a ctrl-c to copy the settings, move your mouse over the image to paste to and then do a ctrl-v to paste.  You could scroll the filmstrip, pausing at each image you wish to paste to, hit crtl-v, then scroll on.  No clicking to select, ctrl-click or shift-click to expand.

**Comment by maboleth** [comment]
> However, have you thought about using the act on behavior. Using it you could hover over an image in the film strip, do a ctrl-c to copy the settings, move your mouse over the image to paste to and then do a ctrl-v to paste. You could scroll the filmstrip, pausing at each image you wish to paste to, hit crtl-v, then scroll on. No clicking to select, ctrl-click or shift-click to expand.

Yes, thanks, that's really useful on the go. I see the purpose.
However, when I have lots of images, it becomes tedious. Since I do photography for a living, I have sets of similar images with the same parameters in a row. 
So usually applying the same settings to a selection of neighboring images is easier than hovering each and every one.

I also have a tendency to refine my edits so I leave the selection for a while, editing one of those images and then doing a quick C/P to all. It doesn't matter which one I edit of those selected as they all have the same parameters.

I explained how I work in Adobe LR, I was trying to use DT in my work too.

If that's hard to change or requires some serious work, I still think DT should give you a warning when you try to Paste on nothing, like "Please move your mouse cursor to a filmstrip before pasting".

**Comment by pehar1** [comment]
> I still think DT should give you a warning when you try to Paste on nothing, like "Please move your mouse cursor to a filmstrip before pasting".

I think "pasting on nothing" is not even possible. Try this scenario:

- have **one** image selected in darkroom (highlighted in filmstrip)
- do CTRL-SHIFT-C or CTRL-C on this image (cut)
- move to the next image in filmstrip (press SPACE), now this second image is highlighted
- hover your mouse cursor anywhere **outside the thumbnails of the filmstrip** (central image area, left panel, right panel, top panel, bottom panel, ...)
- do CTRL-SHIFT-V or CTRL-V (paste)

The history stack (or part of it) you want to copy is pasted on **the image currently edited in central view**. This also happens if you have two or more images selected in filmstrip. This is the expected behavior with "act on". I have made many attempts to achieve a "paste on nothing". Unless I have missed something, this is not possible.

**Comment by maboleth** [comment]
> The history stack (or part of it) you want to copy is pasted on **the image currently edited in central view**. This also happens if you have two or more images selected in filmstrip. This is the expected behavior with "act on". I have made many attempts to achieve a "paste on nothing". Unless I have missed something, this is not possible.
> 
> Edit: the only way I could find nothing **seems** getting pasted is using the same image as source and destination. The history stack does not change. But if you look at the central view, you see popping up "working...", so obviously pasting happened also in this case.

Yes, sorry, I could have been more precise in explaining.

The problem I have with this DT behaviour is actually the selection.
It seems like Selection is almost worthless or has a low priority unless the mouse pointer is not located in the filmstrip.

I described it as "Paste goes into nothing", but actually when you have a selection of images and you are editing one of those in the selection, if you copy/paste the settings, Paste won't go anywhere - unless your mouse is located in the filmstrip.

And that's my problem here. _It's not logical for me to have Paste working for the current image I am in (even with the mouse pointer away from the filmstrip) while the rest of the Selection gets ignored - unless mouse is pointed on the Filmstrip._

It just seems awkward. I wouldn't mind DT having a hovering or "act on" behaviour, but Selection of images should also have some weight or a priority, regardless of my mouse position. I consciously selected the images for some purpose, so let me copy/paste the settings for that selection, without thinking where my mouse pointer is located.

**Comment by MStraeten** [comment]
> Selection of images should also have some weight or a priority, regardless of my mouse position.

There’s no way to give both selection and ‘act on’ the same highest priority. If an image (active image in the central view or a hovered image in the filmstrip) gets the ‘act on’ focus then the selection has minor priority.

**Comment by ralfbrown** [comment]
Please put any continued discussion on #16850.
